elementary/monitor/alerts/group_of_alerts.py

```diff
@@ -0,0 +1,334 @@
+
+        return self._message_builder.get_slack_message()
+
+    def _title_block(self):
+        title = f":small_red_triangle: {self._title}"
+        return title
+
+    def _number_of_failed_block(self):
+        # small_red_triangle: Falied: 36    |    :Warning: Warning: 3    |    :exclamation: Errors: 1
+        fields = []
+        all_components = list(self._components_to_alerts.items())
+        all_components_but_last = all_components[:-1]
+        for component, alert_list in all_components_but_last:
+            fields.append(
+                f":{component.emoji_in_summary}: {component.name_in_summary}: {len(alert_list)}    |"
+            )
+        component, alert_list = all_components[-1]
+        fields.append(
+            (
+                f":{component.emoji_in_summary}: {component.name_in_summary}: {len(alert_list)}"
+            )
+        )
+
+        return fields
+
+    def _get_banner_block(self, env):
+        return None
+
+    def _get_model_error_block_header(self) -> List:
+        model_error_alert_list = self._components_to_alerts[ModelErrorComponent]
+        if len(model_error_alert_list) == 0:
+            return []
+        result = []
+        for model_error_alert in model_error_alert_list:
+            if model_error_alert.message:
+                result.extend(
+                    ["*Result message*"]
+                )
+        return result
+
+    def _get_model_error_block_body(self) -> str:
+        model_error_alert_list = self._components_to_alerts[ModelErrorComponent]
+        if len(model_error_alert_list) == 0:
+            return ""
+        for model_error_alert in model_error_alert_list:
+            if model_error_alert.message:
+                return f"```{model_error_alert.message.strip()}```"
+        return ""
+
+
+    def _attention_required_blocks(self):
+        preview_blocks = [
+            f"*{self._db}.{self._schema}.{self._model}*"
+        ]
+
+        for component, val in self._components_to_attention_required.items():
+            text = f"_{component.empty_section_content}_" if not val else val
+            preview_blocks.append(
+                f"*{component.name_in_summary}*: {text}"
+            )
+
+        return preview_blocks
+
+    def _tabulate_list_of_alerts(self, alert_list):
+        ret = []
+        for alert in alert_list:
+            ret.append(self._get_tabulated_row_from_alert(alert))
+        return ret
+
+    def _get_tabulated_row_from_alert(self, alert: Alert):
+        raise NotImplementedError
+
+    def _had_channel_clashes(self):
+        return False
+
+    def _get_title(self):
+        return None
+
+
+class GroupOfAlertsByTable(GroupOfAlerts):
+    def __init__(
+            self, alerts: List[Alert], default_channel_destination: str, env: str = "dev"
+    ):
+
+        # sort out model unique id
+        models = set([alert.model_unique_id for alert in alerts])
+        if len(models) != 1:
+            raise ValueError(
```

Comment:
I am not sure if we want to fail the all alert flow for having one un-parse-able alert.
Alerts should always run and we need to have a fallback when alert is broken

Reply:
keeping this for now